compiler/rustc_borrowck/src/type_check/constraint_conversion.rs

```diff
@@ -150,7 +158,13 @@ impl<'a, 'tcx> ConstraintConversion<'a, 'tcx> {
             }
 
             let mut next_outlives_predicates = vec![];
-            for (ty::OutlivesPredicate(k1, r2), constraint_category) in outlives_predicates {
+            for (pred, constraint_category) in outlives_predicates {
+                // Constraint is implied by a coroutine's well-formedness.
+                if higher_ranked_assumptions.contains(&pred) {
+                    continue;
+                }
```

Comment:
This is kind of the "main action" in this PR -- right? i.e., when we expand auto-traits, we add to the infcx, then later here, borrowck gets to ignore things.